tracee-ebpf/tracee/tracee.go

```diff
@@ -616,212 +619,291 @@ func (t *Tracee) setUintFilter(filter *UintFilter, filterMapName string, configF
 	if !filter.Enabled {
 		return nil
 	}
+	filterEqualC := C.uint(filterEqual)
+	filterNotEqualC := C.uint(filterNotEqual)
 
+	// equalityFilter filters events for given maps:
+	// 1. uid_filter        u32, u32
+	// 2. pid_filter        u32, u32
+	// 3. mnt_ns_filter     u64, u32
+	// 4. pid_ns_filter     u64, u32
 	equalityFilter, err := t.bpfModule.GetMap(filterMapName)
 	if err != nil {
 		return err
 	}
 	for i := 0; i < len(filter.Equal); i++ {
 		if filter.Is32Bit {
-			err = equalityFilter.Update(uint32(filter.Equal[i]), filterEqual)
+			EqualC := C.uint(filter.Equal[i])
+			err = equalityFilter.Update(unsafe.Pointer(&EqualC), unsafe.Pointer(&filterEqualC))
```

Comment:
I think it should be ok if you just give unsafe.Pointer(&StackID) and remove the "import C" part, same for all other places below.
This is exactly how the old libbpfgo code worked, and it was fine.